cmd/version.go

```diff
@@ -16,11 +16,64 @@ package cmd
 
 //go:generate ./mkversion.sh
 
-// Version will be overwritten at build-time via mkversion.sh
-var Version = "unknown"
+import (
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+)
 
-func MockVersion(version string) (restore func()) {
-	old := Version
-	Version = version
-	return func() { Version = old }
+var (
+	// Version will be overwritten at build-time via mkversion.sh
+	Version = "unknown"
+
+	pid2ProcPath = "/proc/2/status"
```

Comment:
This file is called "version.go". It should remain unchanged, all this should go into its own file.

Reply:
Resolved. Placed in ```detect.go```.